tests/page.tests

```diff
@@ -6,11 +6,13 @@ Raising request handler
 addRoute("/", fun(_) { receive { case _ -> raise } } )
 stdout : () : ()
 args : --enable-handlers --session-exceptions
+ignore : Not clear whether this should be allowed or not
 
 Raising request handler and error handler
 addLocatedRouteWithErrors("/", fun (_, _) { receive { case Foo -> raise } }, fun (_, _, _) { receive { case Bar -> raise } })
 stdout : () : ()
 args : --enable-handlers --session-exceptions
+ignore : Not clear whether this should be allowed or not
```

Comment:
Nope, certainly not now that `SessionFail` manifests itself in the type -- see the signature in `prelude.links` which has been restricted (by @dhil a while back) to only allow the `hear` effect. This will have been broken by the more precise type you have given to `raise` (which is the correct thing to do).

I guess the question is whether we add `SessionFail` to the type signature of `addRoute` and others -- but I think not, probably.